infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftPods.java

```diff
@@ -205,20 +223,75 @@ public void onClose(KubernetesClientException ignored) {}
         throw new InfrastructureException(ex.getMessage());
       }
     }
-    handlers.add(handler);
+    podActionHandlers.add(handler);
+  }
+
+  /**
+   * Registers a specified handler for handling events about changes in pods containers.
+   *
+   * @param handler pod container events handler
+   * @throws InfrastructureException if any error occurs while watcher starting
+   */
+  public void watchContainers(ContainerEventHandler handler) throws InfrastructureException {
```

Comment:
This class gets bigger. Consider moving some code to a separate class, probably watching code

Reply:
Makes sense. I'll consider it.